### ISSUE TITLE
When did the bus actually called at a stop?

### ISSUE THREAD

**Issue by daphshez** [hasadna/open-bus#11]
This is the next task that we will need to tackle once we build up some data in the SIRI database. 

SIRI data only gives estimate for the next arrival to each stop. What we should see (if we poll frequently enough), is a bus arrival time going down monotonously, then the bus at the stop, then the time jumping up. 

In practice it is going to probably be more complicated than that because:
1) The algorithm that estimates next arrival is VERY often wrong
2) Our planned polling time is currently set to 5 minutes. 

Once we accumulate enough data, we need to analyse it and see how best to estimate calling time.

**Comment by daphshez** [comment]
So, good news.
We have a few days of data for the stops of three bus lines: 
- 19 in Jerusalem (from Ein Karem to Har Hazofim) - 40 stops 
- 480 from Jerusalem to Tel-Aviv - 3 stops
- 947 from Haifa to Jerusalem - 29 stops 

Because we poll stops (not lines) we also have data for all other buses that stop at the same stops. 

The data is accessible through [re:dash](http://data.obudget.org), in the arrivals table.

The fields of the table aren't self explanatory, 
so you may need to [read the doc](http://he.mot.gov.il/index.php?option=com_content&view=article&id=2243:pub-trn-memchakim&catid=167:pub-trn-dev-info&Itemid=304). If you do, how about summarising the relevant parts for everyone else?

We also need to start looking the the data. Does it make sense at all? There may be problems with the script the fetches the data and inserts to the db...

**Comment by OrLichter** [comment]
Hi,
I made a short description of what I could make of the tables that we got from siri.
From what I understand that's not all tables. Is there somewhere I could get at least the names of the tables available? I quickly scanned the doc, but it's more about the connection between different parts of the process and not about what's available there.

https://docs.google.com/spreadsheets/d/1Hi2DvVzpz_LVUMEBZ0qMOxp4jyXcRF4jlojK5Uioey8/edit?usp=sharing

**Comment by daphshez** [comment]
@OrLichter ,
The only tables coming from Siri are arrivals & responses. 
Response is the unprocessed xml. Arrivals is the the parsed response. 
You can see the script that generated them here: https://github.com/hasadna/open-bus/blob/master/siri/fetch_and_store_arrivals.py

The rest of the tables are less relevant at his point, but here's some info. 

agency, calendar, routes, shapes, stop_times, stops & trips are the GTFS table. [There's some info and links about the GTFS here](https://github.com/hasadna/open-bus/blob/master/doc/working_with_GTFS.md). 

route_story_stops and trip_route_stories are generated by [this script](https://github.com/hasadna/open-bus/blob/master/gtfs/parser/route_stories.py), the doc string can give you an idea of what it's meant to do. But I don't think it's necessary for this task. 

train_station_exits & station_walking_distance were generated as part [of this issue](https://github.com/hasadna/open-bus/issues/7).

**Comment by daphshez** [comment]
Amir Carmel of the MoT suggested a method for finding bus locations. I am not posting it here because GitHub masses up RTL, but it's available on the mailing list here:

https://groups.google.com/forum/#!topic/openbus-israel/KC60TZu1GyI

I haven't made much effort to understand what he is suggesting. It does seem to me that it might require keeping polling other information than what we currently have in the DB.

**Comment by daphshez** [comment]
Good news: apparently there are some extra fields in the SIRI reply that we haven't been keeping in the past.

In particular `vehicle_ref` is the license number of the bus! This should make tracking the buses much easier. 

Can anyone (@OrLichter? @efratoio?) have a look at the problem again and see if it's now more feasible?

Notes:
- I changed the relevant database tables. The main table is now called `siri_arrivals`.  (There's also `siri_raw_response` that contains the original XML)
- I wrote [documentation about SIRI.](https://github.com/hasadna/open-bus/blob/master/doc/working_with_SIRI.md) I hope it's more readable than the PDFs supplied by the MOT.
